### DISCUSSION TITLE
Meaning of training NNue with Leela Data

### DISCUSSION THREAD

**Discussion by Dboingue** [official-stockfish/Stockfish#4657] [Q&A]
Is there a place in the constellation of github repository or other internet places where the outer information flow of NNue training is explained in term of how it uses Lc0 training data.

Does this mean that the blog from SF12 about a master network being trained from many positions with target to fit being a moderate depth search with classical SF?  I have been following those blogs, and it seems that SF16 is introducing something new, but I find the documentation linked out of that last blog focusing on compressions and optimization of computing costs, which i know is the great innovation in fast NN usage with minimal resolution of scoring quantization and all.  Minimal feature set, just good enough to fit the training target data.. (but that was classical SF scoring on many generated positions in SF12, supervised learning mode).

But as potential user of SF I would like to be able to interpret what it is doing from the input and output point of view and then independent sources of data it is being optimized with.

I don't care much about how fast it does those things, at some documentation level as analytical user of SF eventually. But i would like to be able to interpret what it does to chess information, and what information it embeds.  

I am sorry if this discussion is already on going somewhere.  I would be glad to be forwarded there.. Thanks.

if SF has stopped using classical SF scoring for targets of training of NNue, that would be a major information influx of independent data and models..   It would open progress again not just from ELO gain from speed allowing deeper (which is my understanding of how speed is used by SF, not important).   Perhaps the bits and pieces about WDL conversion i have see briefly while seeking the above information, are about simulating CP from whole games.. 

I am curious about the training set input and output definitions without having to be using internal jargon of the programming functions.  There are existing terminology in machine learning that can be used and that is portable from one application to another.

Is there documentation that is not just for existing dev consumption.. We need to be able to use tools with maximal interpret-ability.
If this is done somewhere.   please let me know. i don,t have the energy to search in the constellation anymore...

**Comment by Dboingue** [comment]
https://github.com/official-stockfish/Stockfish/wiki/Stockfish-FAQ#interpretation-of-the-stockfish-evaluation

makes it even more crucial to tie with that output display from internal scaling ballet across versions.  I am in the process (no ETA)  of reading that, but my fearful pre-conception is that it is only a normalization that will hide the tumult.  That perhaps the leela data is only used partially as input data vector to the master NNue trainin, not also as game outcome target data vector (which is an abuse, as it reased the question of the nature of the training, the supervised.   Lc0 does such SL training of its smaller nets from the RL training, if the NNue master is like one of those, then we now have 2 documentation problems..   Should i go to lc0 documentation, also an open-source multicollaborator evovling project, or has been since I last looked at documentation there.

Or could SF devs do us the favor of another very compact sentence or 2 in its blog post 16 of same caliber as in SF12 Blog. so I don't have to disperse my question back to Lc0..   into how do they interpret or even make their traininng..  I know how..  They use their master network.   Is SF using lc0 iteself as oracle.  that would be a great leap.  and other can of worm interpretation challenge.. 

Edit: can of worm is too strong. I mean something requiring attention and transparency to get users on the train, not left behind.
